File: python/questions/longestConsecutive.py
from typing import List

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if (len(nums) == 0):
            return 0
        ans = 1
        st = set(nums)
        for n in st:
            if n-1 not in st:
                cn = n
                cl = 1
                while cn+1 in st:
                    cn += 1
                    cl += 1
                ans = max(ans, cl)
        return ans

File: rust/questions/longest_consecutive.rs
use std::collections::HashSet;

impl Solution {
    pub fn longest_consecutive(nums: Vec<i32>) -> i32 {
        if nums.is_empty() {
            return 0;
        }
        let st: HashSet<i32> = nums.iter().cloned().collect();
        let mut ans = 1;
        for &n in &st {
            if !st.contains(&(n - 1)) {
                let mut cn = n;
                let mut cl = 1;
                while st.contains(&(cn + 1)) {
                    cn += 1;
                    cl += 1;
                }
                ans = ans.max(cl);
            }
        }
        ans
    }
}
